android/PartialSync/app/src/main/java/io/realm/todo/ItemsActivity.java

```diff
@@ -0,0 +1,127 @@
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.helper.ItemTouchHelper;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.EditText;
+
+import io.realm.Realm;
+import io.realm.Sort;
+import io.realm.SyncUser;
+import io.realm.todo.model.Item;
+import io.realm.todo.model.Project;
+import io.realm.todo.ui.ItemsRecyclerAdapter;
+
+public class ItemsActivity extends AppCompatActivity {
+
+    private Realm mRealm;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_items);
+
+        setSupportActionBar(findViewById(R.id.toolbar));
+
+        String projectId = getIntent().getStringExtra("project_id");
+
+        findViewById(R.id.fab).setOnClickListener(view -> {
+            View dialogView = LayoutInflater.from(this).inflate(R.layout.dialog_task, null);
+            EditText taskText = dialogView.findViewById(R.id.task);
+            new AlertDialog.Builder(ItemsActivity.this)
+                    .setTitle("Add a new task")
+                    .setMessage("What do you want to do next?")
+                    .setView(dialogView)
+                    .setPositiveButton("Add", (dialog, which) -> mRealm.executeTransactionAsync(realm -> {
+                        Item item = new Item();
+                        item.setBody(taskText.getText().toString());
+                        realm.where(Project.class).equalTo("id", projectId).findFirst().getTasks().add(item);
+                    }))
+                    .setNegativeButton("Cancel", null)
+                    .create()
+                    .show();
+        });
+
+        mRealm = Realm.getDefaultInstance();
+        Project project = mRealm.where(Project.class).equalTo("id", projectId).findFirst();
+
+        final ItemsRecyclerAdapter itemsRecyclerAdapter = new ItemsRecyclerAdapter(project.getTasks().sort("timestamp", Sort.ASCENDING));
+        RecyclerView recyclerView = findViewById(R.id.recycler_view);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        recyclerView.setAdapter(itemsRecyclerAdapter);
+
+        ItemTouchHelper.SimpleCallback simpleItemTouchCallback = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT) {
+
+            @Override
+            public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
+                return false;
+            }
+
+            @Override
+            public void onSwiped(RecyclerView.ViewHolder viewHolder, int swipeDir) {
+                int position = viewHolder.getAdapterPosition();
+                String id = itemsRecyclerAdapter.getItem(position).getItemId();
+                mRealm.executeTransactionAsync(realm ->
+                        realm.where(Item.class)
```

Comment:
There is a potential race condition here, if e.g. this item was deleted on the server first. `findFirst` might return null.